node-runner-cli/commands/dockercommand.py

```diff
@@ -0,0 +1,292 @@
+        if run_gateway:
+            detailed_gateway_settings: GatewayDockerSettings = GatewayDockerSettings({}).create_config(
+                postgrespassword)
+            configuration.gateway_settings = detailed_gateway_settings
+            configuration.common_settings.ask_enable_nginx_for_gateway(nginx_on_gateway)
+            config_to_dump["gateway"] = dict(configuration.gateway_settings)
+        else:
+            configuration.common_settings.nginx_settings.protect_gateway = "false"
+
+    if configuration.common_settings.check_nginx_required():
+        configuration.common_settings.ask_nginx_release()
+    else:
+        configuration.common_settings.nginx_settings = None
+
+    config_to_dump["common_config"] = dict(configuration.common_settings)
+
+    yaml.add_representer(type(None), Helpers.represent_none)
+    Helpers.section_headline("CONFIG is Generated as below")
+    print(f"\n{yaml.dump(config_to_dump)}")
+
+    old_config = Docker.load_all_config(config_file)
+    print(f"""
+        {Helpers.section_headline("Differences")}
+        Difference between existing config file and new config that you are creating
+        {dict(DeepDiff(old_config, config_to_dump))}
+          """)
+
+    to_update = ""
+    if autoapprove:
+        print("In Auto mode - Updating file as suggested in above changes")
+    else:
+        to_update = input("\nOkay to update the file [Y/n]?:")
+    if Helpers.check_Yes(to_update) or autoapprove:
+        print(f"\n\n Saving to file {config_file} ")
+
+        with open(config_file, 'w') as f:
+            yaml.dump(config_to_dump, f, default_flow_style=False, explicit_start=True, allow_unicode=True)
+
+
+@dockercommand([
+    argument("-f", "--configfile",
+             help="Path to config file. This file is generated by running `radixnode docker config`"
+                  f"The default value is `{Helpers.get_default_node_config_dir()}/config.yaml` if not provided",
+             default=f"{Helpers.get_default_node_config_dir()}/config.yaml",
+             action="store"),
+    argument("-a", "--autoapprove", help="Pass this option to run without any prompts. "
+                                         "Use this for automation purpose only", action="store_true"),
+    argument("-u", "--update", help="Pass this option to update the deployed softwares to latest version."
+                                    " CLI prompts to confirm the versions if '-a' is not passed",action="store_true")
+])
+def setup(args):
+    """
+    This commands setups up the software and deploys it based on what is stored in the config.yaml file.
+    To update software versions, most of the time it is required to update the versions in config file and run this command
+    """
+    autoapprove = args.autoapprove
+    all_config = Docker.load_all_config(args.configfile)
+    update = args.update
+
+    new_config = Docker.update_versions(all_config, autoapprove) if update else all_config
```

Comment:
If the versions needed to change the configuration, what's the intended action to prevent someone updating with incorrect configuration? 

I guess prompting them to install a newer CLI version; which updates the templates?

Reply:
Idea is to have versions that setup command can support. If there is breaking version change and config is still pointing to old version, the setup command should stop

Reply:
And prompt them to run config or migrate it